src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,83 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private Entry<K, V>[] table;
+    private int size;
+
+    private static class Entry<K, V> {
+        private final K key;
+        private V value;
+        private final Entry<K, V> next;
+
+        private Entry(K key, V value, Entry<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    public MyHashMap() {
+        this.table = new Entry[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= LOAD_FACTOR * table.length) {
+            resize();
+        }
+        int index = hashIndex(key);
+        Entry<K, V> entry = table[index];
+        while (entry != null) {
+            if (Objects.equals(entry.key, key)) {
+                entry.value = value;
+                return;
+            }
+            entry = entry.next;
+        }
```

Comment:
extract it to separate method and reuse it. You have the same code in getValue()

Reply:
Thanks for the comment. But I think it is inappropriate to complicate the code with an additional separate method, because at first glance the code is repeated only twice. But apart from that, it's not really like that, because put and getValue return values accepting different input parameters. And creating and redefining additional variables specifically for this additional method will make the code more difficult to read and performance will deteriorate.